src/util/helpers.js

```diff
@@ -45,3 +51,19 @@ export function wrap(value: any, response?: any) {
   });
   return value;
 }
+
+export function generateAuthorizeUrl(input: OAuthAuthorizeInput): string {
+  const clientId = input.clientId || this.options.clientId;
+  const state = input.state;
+  let redirectUri = input.redirectUri || this.options.redirectUri;
+
+  if (!clientId || !state || !redirectUri) {
+    throw new BaseError(
+      "Client credentials are missing. Please doublecheck clientId, redirectUri and state"
+    );
+  }
+
+  redirectUri = encodeURIComponent(redirectUri);
+
+  return `https://app.abstract.com/signin/oauth/authorize?client_id=${clientId}&redirect_uri=${redirectUri}&response_type=code&scope=all&state=${state}`;
```

Comment:
nit: could inline and avoid re-assignment

```suggestion
  const redirectUri = input.redirectUri || this.options.redirectUri;

  if (!clientId || !state || !redirectUri) {
    throw new BaseError(
      "Client credentials are missing. Please doublecheck clientId, redirectUri and state"
    );
  }

  return `https://app.abstract.com/signin/oauth/authorize?client_id=${clientId}&redirect_uri=${encodeURIComponent(redirectUri)}&response_type=code&scope=all&state=${state}`;
```

Re-assignments and mutations break type refinements. It also has the added benefit of treating the string like a view template 🙂